packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -101,16 +95,30 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
 
         this.term = new Xterm.Terminal({
             experimentalCharAtlas: 'dynamic',
+            experimentalBufferLineImpl: 'TypedArray',
             cursorBlink: false,
-            fontFamily: cssProps.fontFamily,
-            fontSize: cssProps.fontSize,
+            fontFamily: this.preferences['terminal.integrated.fontFamily'],
+            fontSize: this.preferences['terminal.integrated.fontSize'],
+            fontWeight: this.preferences['terminal.integrated.fontWeight'],
+            fontWeightBold: this.preferences['terminal.integrated.fontWeightBold'],
+            letterSpacing: this.preferences['terminal.integrated.letterSpacing'],
+            lineHeight: this.preferences['terminal.integrated.lineHeight'],
             theme: {
                 foreground: cssProps.foreground,
                 background: cssProps.background,
                 cursor: cssProps.foreground,
                 selection: cssProps.selection
             },
         });
+        this.toDispose.push(this.preferences.onPreferenceChanged(change => {
+            const lastSeparator = change.preferenceName.lastIndexOf('.');
+            if (lastSeparator > 0) {
+                const preferenceName = change.preferenceName.substr(lastSeparator + 1);
+                this.term.setOption(preferenceName, change.newValue);
```

Comment:
I think that is not correct resetting the defaults. In the following gif I'm removing the font setting for terminal from preferences and I appears to set an empty value, at least that's how to explain the bogus `monospace` font, which is set in Chrome for testing.

![2019-02-19 10 14 28](https://user-images.githubusercontent.com/914497/53003635-b49ff580-342f-11e9-9f52-c1ae3ea72f20.gif)

Reply:
Oh I would have expected to get the default in in that case.

Reply:
fixed it